spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/actions/TestSnapshotTableAction.java

```diff
@@ -65,4 +69,40 @@ public void testSnapshotWithParallelTasks() throws IOException {
         .execute();
     assertThat(snapshotThreadsIndex.get()).isEqualTo(2);
   }
+
+  @TestTemplate
+  public void testTableLocationOverlapThrowsException() throws IOException {
+    // Ensure the test runs only for non-Hadoop-based catalogs,
+    // because path-based tables cannot have a custom location set.
+    assumeThat(catalogName)
+        .as("Cannot set a custom location for a path-based table.")
+        .isNotEqualTo("testhadoop");
+
+    String location = Files.createTempDirectory(temp, "junit").toFile().toString();
+
+    sql(
+        "CREATE TABLE %s (id bigint NOT NULL, data string) USING parquet LOCATION '%s'",
+        SOURCE_NAME, location);
+    sql("INSERT INTO TABLE %s VALUES (1, 'a')", SOURCE_NAME);
+    sql("INSERT INTO TABLE %s VALUES (2, 'b')", SOURCE_NAME);
+
+    // Define properties for the destination table, setting its location to the same path as the
+    // source table
+    Map<String, String> tableProperties = Maps.newHashMap();
+    tableProperties.put("location", "file:" + location);
+
+    // Test that an exception is thrown
+    // when the destination table location overlaps with the source table location
+    // Assert that the exception message matches the expected error message
```

Comment:
> Assert that the exception message matches the expected error message

I would remove this comment. This is obvious when reading the following code.

Reply:
Thank you for the suggestion. I have removed this part of the comment.